network/src/peer_manager/mod.rs

```diff
@@ -816,24 +829,44 @@ where
     ) {
         let mut upstream_handlers = self.upstream_handlers.clone();
         let network_context = self.network_context.clone();
+        let inbound_rate_limiter = self.inbound_rate_limiter.clone();
         self.executor.spawn(network_events.for_each_concurrent(
             self.max_concurrent_network_reqs,
             move |inbound_event| {
                 Self::handle_inbound_event(
                     network_context.clone(),
                     inbound_event,
                     peer_id,
+                    inbound_rate_limiter.clone(),
```

Comment:
you clone, so that you can clone? what is this weirdness? same with network context :O

does this have something to do with self not being within the async context?

Reply:
This has to deal with the `for_each_concurrent` then `move` :(